sensu/transport/rabbitmq/rabbitmq_config.go

```diff
@@ -0,0 +1,80 @@
+package rabbitmq
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/upfluence/goutils/log"
+)
+
+var errNoUserInURI = errors.New("No user found in URI")
+
+type TransportConfig struct {
+	Host      string      `json:"host,omitempty"`
+	Port      json.Number `json:"port,omitempty"`
+	Vhost     string      `json:"vhost,omitempty"`
+	User      string      `json:"user,omitempty"`
+	Password  string      `json:"password,omitempty"`
+	Heartbeat json.Number `json:"heartbeat,omitempty"`
+	Prefetch  json.Number `json:"prefetch,omitempty"`
+	Ssl       struct {
+		CertChainFile  string `json:"cert_chain_file,omitempty"`
+		PrivateKeyFile string `json:"private_key_file,omitempty"`
+	} `json:"ssl,omitempty"`
+
+	uri string `json:"-"`
+}
+
+func (c *TransportConfig) GetURI() string {
+	if c.uri == "" {
+		c.uri = fmt.Sprintf(
+			"amqp://%s:%s@%s:%s/%s",
+			c.User,
+			c.Password,
+			c.Host,
+			c.Port.String(),
+			url.QueryEscape(c.Vhost),
+		)
+	}
+
+	return c.uri
+}
+
+func NewTransportConfig(uri string) (*TransportConfig, error) {
+	uriComponents, err := url.Parse(uri)
+	if err != nil {
+		log.Errorf("Failed to parse the URI: %s", err)
```

Comment:
is it really useful to log the errors before return them? I do think it would be easier to only log the error returned when you call `NewTransportConfig`